data-integration/main.bal

```diff
@@ -0,0 +1,102 @@
+            if result is persist:AlreadyExistsError {
+                return http:CONFLICT;
+            }
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return http:CREATED;
+    }
+
+    isolated resource function post task(store:EmployeeTaskInsert task)
+            returns http:Created|http:Conflict|http:InternalServerError {
+        int[]|persist:Error result = self.dbClient->/employeetasks.post([task]);
+        if result is persist:Error {
+            if result is persist:AlreadyExistsError {
+                return http:CONFLICT;
+            }
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return http:CREATED;
+    }
+
+    isolated resource function get task/[int taskId]() returns http:Ok|http:NotFound|http:InternalServerError {
+        store:EmployeeTask|persist:Error result = self.dbClient->/employeetasks/[taskId];
+        if result is persist:Error {
+            if result is persist:NotFoundError {
+                return http:NOT_FOUND;
+            }
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return <http:Ok>{body: result};
+    }
+
+    isolated resource function get employeetasks/[int empId]() returns http:Ok|http:NotFound|http:InternalServerError {
+        store:EmployeeTask[]|persist:Error result = from store:EmployeeTask task in self.dbClient->/employeetasks(store:EmployeeTask)
+            where task.employeeId == empId
+            select task;
+        if result is persist:Error {
+            if result is persist:NotFoundError {
+                return http:NOT_FOUND;
+            }
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return <http:Ok>{body: result};
+    }
+
+    isolated resource function get employee/[int empId]() returns http:Ok|http:NotFound|http:InternalServerError {
+        store:Employee|persist:Error result = self.dbClient->/employees/[empId];
+        if result is persist:Error {
+            if result is persist:NotFoundError {
+                return http:NOT_FOUND;
+            }
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return <http:Ok>{body: result};
+    }
+
+    isolated resource function put task/[int taskId](store:EmployeeTaskUpdate emp)
+            returns http:Ok|http:InternalServerError {
+        store:EmployeeTask|persist:Error result = self.dbClient->/employeetasks/[taskId].put(emp);
+        if result is persist:Error {
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return <http:Ok>{body: result};
+    }
+
+    isolated resource function delete task/[int taskId]()
+            returns http:NoContent|http:NotFound|http:InternalServerError {
+        store:EmployeeTask[]|persist:Error result = from store:EmployeeTask task in self.dbClient->/employeetasks(store:EmployeeTask)
```

Comment:
Isn't there just one task with this task ID? Why can't we directly delete or get by ID, check the status, and delete?